warehouse/accounts/services.py

```diff
@@ -75,6 +78,19 @@ def check_password(self, userid, password):
 
         return True
 
+    def create_user(self, user):
+        if self.find_userid(user.username):
+            raise UserAlreadyExists(
+                "User with username {0} already exists!".format(user.username)
+            )
+        self.db.add(user)
+        self.db.commit()
+
+    def update_user(self, user):
+        user = self.db.merge(user)
+        self.db.add(user)
+        self.db.commit()
```

Comment:
Similar to above, we shouldn't commit here, transactions are managed at a higher level.